docs/docs/reference/witnesses/witnesses.md

```diff
@@ -0,0 +1,186 @@
+
+## Witnesses for Extension Methods
+
+Witnesses can also be defined without a `for` clause. A typical application is to use a witness to define extension methods. Examples:
+
+```scala
+witness StringOps {
+  def longestStrings(this xs: Seq[String]): Seq[String] = {
+    val maxLength = xs.map(_.length).max
+    xs.filter(_.length == maxLength)
+  }
+}
+
+witness ListOps {
+  def second[T](this xs: List[T]) = xs.tail.head
+}
+```
+Witnesses like these translate to `implicit` objects without an extends clause.
+
+## Anonymous Witnesses
+
+The name of a witness definition can be left out. Examples:
+```scala
+witness for Ord[Int] { ... }
+witness [T: Ord] for Ord[List[T]] { ... }
+
+witness {
+  def second[T](this xs: List[T]) = xs.tail.head
+}
+```
+If the name of a witness is missing, the compiler will synthesize a name from
+the type in the for clause, or, if that is missing, from the first defined
+extension method. Details remain to be specified.
+
+## Conditional Witnesses
+
+A witness can depend on another witness being defined. For instance:
+```scala
+trait Convertible[From, To] {
+  def convert (this x: From): To
+}
+
+witness [From, To] with (c: Convertible[From, To]) for Convertible[List[From], List[To]] {
+  def convert (this x: List[From]): List[To] = x.map(c.convert)
+}
+```
+
+The `with` clause in a witness defines required witnesses. The witness for `Convertible[List[From], List[To]]` above is defined only if a witness for `Convertible[From, To]` exists.
+`with` clauses translate to implicit parameters of implicit methods. Here is the expansion of the anonymous witness above in terms of a class and an implicit method (the example demonstrates well the reduction in boilerplate that witness syntax can achieve):
+```scala
+class Convertible_List_List_witness[From, To](implicit c: Convertible[From, To])
+extends Convertible[List[From], List[To]] {
+  def convert (this x: List[From]): List[To] = x.map(c.convert)
+}
+implicit def Convertible_List_List_witness[From, To](implicit c: Convertible[From, To])
+  : Convertible[List[From], List[To]] =
+  new Convertible_List_List_witness[From, To]
+```
+Context bounds in witness definitions also translate to implicit parameters, and therefore they can be represented alternatively as with clauses. For instance, here is an equivalent definition of the `ListOrd` witness:
+```scala
+witness ListOrd[T] with (ord: Ord[T]) for List[Ord[T]] { ... }
+```
+An underscore ‘_’ can be used as the name of a required witness, if that witness does not
+need to be referred to directly. For instance, the last `ListOrd` witness could also have been written like this:
+```scala
+witness ListOrd[T] with (_: Ord[T]) for List[Ord[T]] { ... }
```

Comment:
I believe it should be `Ord[List[T]]` instead of `List[Ord[T]]`. (save as above)